src/Components/Space.tsx

```diff
@@ -191,6 +195,37 @@ const Space = memo(({
     }
   }
 
+  const rotateArray = (array: any[] ) => {
+    const newArray = array.slice(1);
+    newArray.push(array[0]);
+    return newArray;
+  }
+
+  const rotateAbilities = async (players: DBPlayer[]) => {
+
+    if (players.length === 1) return;
+    else {
+      const updatedPlayers = [...players]
+      const abilitiesArray = players.map(player => player.playerAbilities);
+      const directionsArray = players.map(player => player.playerDirections);
+
+      const rotatedAbilitiesArray = rotateArray(abilitiesArray);
+      const rotatedDirectionsArray = rotateArray(directionsArray);
+      console.log("abilities array", abilitiesArray)
+      console.log("rotated Abilities",rotatedAbilitiesArray);
+
+      updatedPlayers.forEach((player, i) => {
+        player.playerAbilities = rotatedAbilitiesArray[i];
+        player.playerDirections = rotatedDirectionsArray[i];
+      })
+
+      console.log("updatedPlayers", updatedPlayers)
+
+      await setDoc(gameState.roomId, {
+        players: updatedPlayers
```

Comment:
here you can add
note: make sure you pass down the existing `updateAbilitiesCount` value from `newRoomValue`
```
updateAbilitiesCount: updateAbilitiesCount + 1
```